marlowe/specification/marlowe-cardano-specification.md

```diff
@@ -177,14 +177,23 @@ Furthermore, let `marloweValidatorHash :: ValidatorHash` be the hash of the Marl
 The validation fails (via returning `False` for `validationResult` or via the throwing of an error) if any of the following constraints does not hold. Fundamentally, all failures in Plutus are calls to `error` and all successes are returns of `()`. The typed Plutus validators obscure this distinction slightly by using `Bool` as the return type.
 
 
+#### *Constraint 0.* Input to contract
+
+The input present in the redeemer and script context is the input provided to the semantics computation.
+```haskell
+marloweInput ∪ txInfoValidRange (scriptContextTxInfo scriptContext) ∪ txInfoData (scriptContextTxInfo scriptContext) ≅ transactionInput
+```
+This is a congruence because `marloweInput` contains the input (perhaps including the Merkle hash of the contract's continuation) and `txInfoValidRange` of the script context contains the validity interval for the transaction, whereas `transactionInput` contains both. Additionally, any continued contract is present in `txInfoData` of the script context, whereas it is directly present in any `MerkleizedInput` of `transactionInput`. Also note that the script context may, in principle, contain an open, closed, or half-open/half-closed time interval, but Marlowe semantics requires a closed time interval.
```

Comment:
This `Additionally` sounds like it is a caveat? Should it say `However` intead? Also, is `continued contract` just in the case of merkleized continuations?  If that is the case would it be clearer to say `merkleized contract continuation`. If not then we could say `contract continuation independently of whether it is or not merkleized` to be clearer

Reply:
Thanks! Fixed in 3aaf4563a9db58366c2c7b56069af6313c57cb36.